### PR TITLE
additional options for uvision5 uvoptx generation

### PR DESCRIPTION
**Pull request by ohagendorf**
please see project-generator/project_generator#415

### CODE REVIEW

**Comment by 0xc0170** [comment]
Is `TargetDriverDllRegistry_SetRegEntry_Name` the missing piece ? For flashing some MCU? I cant find it in the projects I have generated for some MCU (not ST though, haven't yet tested it there).

How to get it? We provide parser, and  if this is not part of uvproj, we might need to  rewrite some logic a bit. It accepts one template , in this case `uvproj`. Therefore the parser should accept 2 in this case? 

I need more information for this changeset

File: project_generator_definitions/mcu/st/stm32f401xe.yaml [hunk 1]
```diff
@@ -37,6 +37,8 @@ tool_specific:
             - null
             FlashDriverDll:
             - UL2CM3(-S0 -C0 -P0 -FD20000000 -FC1000 -FN1 -FF0STM32F4xx_512 -FS08000000 -FL080000 -FP0($$Device:STM32F401RETx$CMSIS\Flash\STM32F4xx_512.FLM))
+            TargetDriverDllRegistry_SetRegEntry_Name:
+            - -U0672FF484849785087111542 -O78 -S0 -C0 -A0 -N00("ARM CoreSight SW-DP") -D00(2BA01477) -L00(0) -TO18 -TC10000000 -TP21 -TDS8007 -TDT0 -TDC1F -TIEFFFFFFFF -TIP8 -FO7 -FD20000000 -FC1000 -FN1 -FF0STM32F4xx_512.FLM -FS08000000 -FL080000 -FP0($$Device:STM32F401RETx$CMSIS\Flash\STM32F4xx_512.FLM)
```
Comment (0xc0170) [inline project_generator_definitions/mcu/st/stm32f401xe.yaml]: As I shared on progen, lets discuss this changeset here. How do they differ , line 39 and 41?

I assume -T is trace info,which we dont care about, -F are flas and they match, thus could be copied from flashdriverdll. If we do this, can you flash nucleo properly ? I dont have now with me any nucleo to test :(

Another question: if we remove this `TargetDriverDllRegistry_SetRegEntry_Name`, and generate uvoptx , does it still fail to flash? is there really some value in this ?

Comment (ohagendorf) [inline project_generator_definitions/mcu/st/stm32f401xe.yaml]: Last question first: with my PR immediate flashing is possible: just export, unzip, open in keil, compile and flash - not necessary to open several config dialogs ... before flashing as before.
The shown minimal uvoptx file is enough for this new behaviour.

For me they differ too much that's why I didn't had a deeper look. But your idea was great. The existing FlashDriverDll option used in uvoptx works! I just tried it manually and used exactly the same option for a STM32F446RE. So maybe we don't need any new options except the debugger options nTsel, pMon and Key. I'll check it with more targets and debuggers. 

Unfortunately a nTsel is not enough i.e. without a SetRegEntry part the uvoptx file doesn't work.

Comment (0xc0170) [inline project_generator_definitions/mcu/st/stm32f401xe.yaml]: > Unfortunately a nTsel is not enough i.e. without a SetRegEntry part the uvoptx file doesn't work.

we can generate at least that info, the minimum set. This means we use debugger settings that are generic, and progen definitions are without any changes, that would be great.

Comment (ohagendorf) [inline project_generator_definitions/mcu/st/stm32f401xe.yaml]: Yes, it looks like so, just a few new generic debugger settings (nTsel, ...).

**Comment by ohagendorf** [comment]
I changed everything as dicussed (hopefully)
please see also at project-generator/project_generator#415

**Pull request merged**